graphqlapi/src/main/java/io/stargate/graphql/persistence/schemafirst/SchemaSourceDao.java

```diff
@@ -134,54 +132,33 @@ BoundQuery schemaQuery(String namespace) {
   }
 
   /** @return the new version */
-  public SchemaSource insert(String namespace, UUID expectedLatestVersion, String newContents)
-      throws Exception {
+  public SchemaSource insert(String namespace, String newContents) throws Exception {
 
     ensureTableExists(namespace);
 
     UUID newVersion = Uuids.timeBased();
 
-    BoundQuery latestVersionCas =
-        dataStore
-            .queryBuilder()
-            .update(namespace, TABLE_NAME)
-            .value(LATEST_VERSION_COLUMN_NAME, newVersion)
-            .where(KEY_CONDITION)
-            .ifs(LATEST_VERSION_COLUMN_NAME, Predicate.EQ, expectedLatestVersion)
-            .build()
-            .bind();
-
-    BoundQuery insertNewVersion =
+    BoundQuery insertNewSchema =
         dataStore
             .queryBuilder()
             .insertInto(namespace, TABLE_NAME)
             .value(KEY_COLUMN_NAME, UNIQUE_KEY)
             .value(VERSION_COLUMN_NAME, newVersion)
+            .value(LATEST_VERSION_COLUMN_NAME, newVersion)
             .value(CONTENTS_COLUMN_NAME, newContents)
+            .value(DEPLOYMENT_IN_PROGRESS_COLUMN_NAME, false)
             .build()
             .bind();
 
-    ResultSet resultSet =
-        dataStore.batch(ImmutableList.of(latestVersionCas, insertNewVersion)).get();
-    Row row = resultSet.one();
-    if (!row.getBoolean(APPLIED_COLUMN_NAME)) {
-      UUID actualLatestVersion = row.getUuid(LATEST_VERSION_COLUMN_NAME);
-      String message;
-      if (expectedLatestVersion == null) {
-        message =
-            String.format(
-                "Could not deploy schema: you indicated your changes were the first version,"
-                    + " but there is an existing version '%s'.",
-                actualLatestVersion);
-      } else {
-        message =
-            String.format(
-                "Could not deploy schema: you indicated your changes applied to version '%s',"
-                    + " but there is a more recent version '%s'.",
-                expectedLatestVersion, actualLatestVersion);
-      }
-      throw new IllegalStateException(message);
+    try {
+      dataStore.execute(insertNewSchema).get();
+    } catch (Exception e) {
+      throw new IllegalStateException(
```

Comment:
I am not sure what type of exception we should throw here.

Reply:
GraphQL does not surface the Java type, any runtime exception thrown from a fetcher will be classified as `DataFetchingException`. So it's the message that really matters.

Indeed it's not really an illegal state, but I don't think it's worth creating a custom type either. So maybe simply `RuntimeException`.